src/main/java/org/vanilladb/core/util/Timer.java

```diff
@@ -131,6 +137,17 @@ public long getComponentTime(Object component) {
 		return timer.getTotalTime();
 	}
 
+	// MODIFIED: Add method to record given time
+	public void recordTime(Object component, long time) {
+		SubTimer timer = subTimers.get(component);
+		if (timer == null) {
+			timer = new SubTimer();
+			timer.setTimer(time);
+			subTimers.put(component, timer);
+			componenents.add(component);
+		}
+	}

```

Comment:
Please use `timeInterval`  as the function input. 
Line 145 should be put at the end of this function.
```java
public void recordTime(Object component, long timeInterval) {
	SubTimer timer = subTimers.get(component);
	if (timer == null) {
		timer = new SubTimer();
		subTimers.put(component, timer);
		componenents.add(component);
	}
	timer.setTime(timeInterval);
	}
```
And please move this method to before `getComponentTime`.